scripts/install/nextcloud.sh

```diff
@@ -13,218 +13,166 @@
-        # in all major browsers and getting removed from this list
-        # could take several months.
-        add_header Referrer-Policy "no-referrer" always;
-        add_header X-Content-Type-Options "nosniff" always;
-        add_header X-Download-Options "noopen" always;
-        add_header X-Frame-Options "SAMEORIGIN" always;
-        add_header X-Permitted-Cross-Domain-Policies "none" always;
-        add_header X-Robots-Tag "none" always;
-        add_header X-XSS-Protection "1; mode=block" always;
-
-        # Optional: Don't log access to assets
-        access_log off;
-    }
-
-    location ~ ^\/nextcloud\/.+[^\/]\.(?:png|html|ttf|ico|jpg|jpeg|bcmap)\$ {
-        try_files \$uri /nextcloud/index.php\$request_uri;
-        # Optional: Don't log access to other assets
-        access_log off;
-    }
+function _bootstrap() {
+    echo_progress_start "Configuring Nextcloud settings"
+
+    _occ "maintenance:install -n --database 'mysql' --database-name 'nextcloud'  --database-user 'nextcloud' --database-pass '$nextcldMySqlPW' --admin-user '$masteruser' --admin-pass '$masterpass' "
+    _occ "maintenance:mode --on"
+    _occ "db:add-missing-indices"
+    _occ "db:convert-filecache-bigint --no-interaction"
+    # _occ "config:system:set memcache.local --value='\OC\Memcache\APCu" # TODO follow https://github.com/nextcloud/server/issues/24567
+
+    i=1
+    _occ "config:system:set trusted_domains $i --value='localhost'"
+    ((i++))
+    ip=$(ip route get 1 | sed -n 's/^.*src \([0-9.]*\) .*$/\1/p')
+    _occ "config:system:set trusted_domains $i --value=$ip"
+    ((i++))
+    _occ "config:system:set trusted_domains $i --value=$(hostname)"
+    ((i++))
+    _occ "config:system:set trusted_domains $i --value=$(hostname)"
+    for value in $(grep server_name /etc/nginx/sites-enabled/default | cut -d' ' -f 4 | cut -d\; -f 1); do
+        if [[ $value != "_" ]]; then
+            _occ "config:system:set trusted_domains $i --value=$value"
+            ((i++))
+        fi
+    done
+    #All users but the master user
+
+    _occ "maintenance:mode --off"
+    restart_php_fpm
+    systemctl restart nginx
+    echo_progress_done "Nextcloud configured"
 }
-EOF
-    echo_progress_done
 
-    echo_progress_start "Configuring database"
-    mysql --user="root" --password="$password" --execute="CREATE DATABASE nextcloud;"
-    mysql --user="root" --password="$password" --execute="CREATE USER nextcloud@localhost IDENTIFIED BY '$nextpass';"
-    mysql --user="root" --password="$password" --execute="GRANT ALL PRIVILEGES ON nextcloud.* TO nextcloud@localhost;"
-    mysql --user="root" --password="$password" --execute="FLUSH PRIVILEGES;"
-    echo_progress_done "Database configured"
+function _users() {
+    for u in "${users[@]}"; do
+        echo_progress_start "Adding $u to Nextcloud"
+        OC_PASS=$(_get_user_password "$u")
+        export OC_PASS
+        #TODO decide what happens wih the stdout from this
```

Comment:
Log should be fine? Otherwise, if it's useful info, can we just save the output as a variable and pipe it to an echo?

Reply:
All `_occ` output now gets piped to the log, must have been here as a remnant of when it wasn't